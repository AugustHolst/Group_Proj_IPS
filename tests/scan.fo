fun int incr(int a, int b) = a + b

fun int writeInt(int n) = write(n)

fun [int] main() =
  let a = {1, 2, 3} in
  let b = scan(incr, 0, a) in
  let k = map(writeInt, b) in

  let c = {5, 22, 1000} in
  let d = scan(incr, 0, c) in
  map(writeInt, d)
