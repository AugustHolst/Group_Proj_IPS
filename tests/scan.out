1 3 6 5 27 1027 
